crates/mempool/src/lib.rs

```diff
@@ -0,0 +1,274 @@
+// Copyright 2024 PRAGMA
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashSet;
+use pallas_primitives::conway::{TransactionInput, Tx};
+use amaru_ledger::store::{Snapshot};
+
+pub trait Mempool {
+    type Tx;
```

Comment:
making this generic over Tx / Block is probably not super valuable, IMO

Reply:
Especially if the module still depends on `pallas_primitives::conway::Tx` 😅